diff -rupN --no-dereference med-5.0.0/python/med_enumtest_typemap.i med-5.0.0-new/python/med_enumtest_typemap.i
--- med-5.0.0/python/med_enumtest_typemap.i	2023-03-07 15:20:57.000000000 +0100
+++ med-5.0.0-new/python/med_enumtest_typemap.i	2024-10-25 10:22:44.993666188 +0200
@@ -128,7 +128,7 @@ public:
   if (pclass == NULL) printf("%s\n","Can't get class $1_basetype");
   Py_DECREF(pmod);
   pargs  = Py_BuildValue("(i)",*$1);
-  pinst  = PyEval_CallObject(pclass, pargs);
+  pinst  = PyObject_CallObject(pclass, pargs);
   if (pinst == NULL) printf("%s\n","Can't instanciate class $1_basetype");
   $result=SWIG_Python_AppendOutput($result, pinst);
 }
diff -rupN --no-dereference med-5.0.0/python/med_enum_typemap.i med-5.0.0-new/python/med_enum_typemap.i
--- med-5.0.0/python/med_enum_typemap.i	2023-03-07 15:20:57.000000000 +0100
+++ med-5.0.0-new/python/med_enum_typemap.i	2024-10-25 10:22:44.993666188 +0200
@@ -109,7 +109,7 @@ Type.__repr__= lambda self: #Type +"("+s
   if (pclass == NULL) printf("%s\n","Can't get class $1_basetype");
   Py_DECREF(pmod);
   pargs  = Py_BuildValue("(i)",*$1);
-  pinst  = PyEval_CallObject(pclass, pargs);
+  pinst  = PyObject_CallObject(pclass, pargs);
   if (pinst == NULL) printf("%s\n","Can't instanciate class $1_basetype");
   $result=SWIG_Python_AppendOutput($result, pinst);
 }
