diff -rupN med-5.0.0/python/med_array_typemap.i med-5.0.0-new/python/med_array_typemap.i
--- med-5.0.0/python/med_array_typemap.i	2023-03-07 15:20:57.000000000 +0100
+++ med-5.0.0-new/python/med_array_typemap.i	2024-10-25 10:34:21.291192197 +0200
@@ -181,7 +181,7 @@ Type.__repr__= lambda self: #Type +"("+s
 //  TypeMed * const ParamName : OUT 2/4 (l'allocation Type est faite ds Python)
 %typemap(freearg) TypeMed * const ParamName {
   Py_INCREF(o$argnum);
-  $result=SWIG_Python_AppendOutput($result, o$argnum);
+  $result=SWIG_AppendOutput($result, o$argnum);
 }
 // TypeMed * const (OUT) 3/4
 // pour ne pas activer un out du TypeMed * const (par sécurité)
@@ -290,7 +290,7 @@ Type.__repr__= lambda self: #Type +"("+s
 //  unsigned char * const : OUT 2/4 (l'allocation Type est faite ds Python)
 %typemap(freearg) unsigned char * const {
   Py_INCREF(o$argnum);
-  $result=SWIG_Python_AppendOutput($result, o$argnum);
+  $result=SWIG_AppendOutput($result, o$argnum);
 }
 // unsigned char * const (OUT) 3/4
 // pour ne pas activer un out du unsigned char * const (par sécurité)
diff -rupN med-5.0.0/python/med_bool_typemap.i med-5.0.0-new/python/med_bool_typemap.i
--- med-5.0.0/python/med_bool_typemap.i	2023-03-07 15:20:57.000000000 +0100
+++ med-5.0.0-new/python/med_bool_typemap.i	2024-10-25 10:34:22.137331627 +0200
@@ -22,7 +22,7 @@
    /*      Py_DECREF(o2); */
    /*      Py_DECREF(o3); */
    /*  } */
-  $result=SWIG_Python_AppendOutput($result, o);
+  $result=SWIG_AppendOutput($result, o);
 }
 
 %typemap(in,numinputs=0) med_bool *(med_bool temp) {
diff -rupN med-5.0.0/python/med_enumtest_typemap.i med-5.0.0-new/python/med_enumtest_typemap.i
--- med-5.0.0/python/med_enumtest_typemap.i	2024-10-25 10:22:45.479664799 +0200
+++ med-5.0.0-new/python/med_enumtest_typemap.i	2024-10-25 10:34:41.871593880 +0200
@@ -130,7 +130,7 @@ public:
   pargs  = Py_BuildValue("(i)",*$1);
   pinst  = PyObject_CallObject(pclass, pargs);
   if (pinst == NULL) printf("%s\n","Can't instanciate class $1_basetype");
-  $result=SWIG_Python_AppendOutput($result, pinst);
+  $result=SWIG_AppendOutput($result, pinst);
 }
 
 %typemap(in,numinputs=0) TypeEnum * (TypeEnum temp) {
diff -rupN med-5.0.0/python/med_enum_typemap.i med-5.0.0-new/python/med_enum_typemap.i
--- med-5.0.0/python/med_enum_typemap.i	2024-10-25 10:22:45.479664799 +0200
+++ med-5.0.0-new/python/med_enum_typemap.i	2024-10-25 10:34:31.715912728 +0200
@@ -111,7 +111,7 @@ Type.__repr__= lambda self: #Type +"("+s
   pargs  = Py_BuildValue("(i)",*$1);
   pinst  = PyObject_CallObject(pclass, pargs);
   if (pinst == NULL) printf("%s\n","Can't instanciate class $1_basetype");
-  $result=SWIG_Python_AppendOutput($result, pinst);
+  $result=SWIG_AppendOutput($result, pinst);
 }
 
 %typemap(in,numinputs=0) TypeEnum * (TypeEnum temp) {
